libs/core/src/lib/date-picker/date-picker.component.ts

```diff
@@ -529,6 +529,18 @@ export class DatePickerComponent<D>
         }
     }
 
+    onTodayButtonClick(): void {
+        const todayDate = this._dateTimeAdapter.today();
+        const todayDateRange: DateRange<D> = { start: todayDate, end: todayDate };
```

Comment:
It is not necessary to create both these variables here, since we'll only end up using one of them